crates/host/src/wasmbus/mod.rs

```diff
@@ -2374,108 +2215,93 @@ impl Host {
-        let matching_instances = instances
-            .iter()
-            .filter(|(instance_annotations, _)| {
-                annotations.iter().all(|(k, v)| {
-                    instance_annotations
-                        .get(k)
-                        .is_some_and(|instance_value| instance_value == v)
-                })
-            })
-            .map(|(_, instance)| instance.clone())
-            .collect::<Vec<Arc<ActorInstance>>>();
-        ensure!(
-            !matching_instances.is_empty(),
-            "no actors with matching annotations found to stop"
-        );
+        actor.calls.abort();
 
-        for instance in matching_instances {
-            instances.remove(&instance.annotations);
-            self.uninstantiate_actor(claims, instance.clone()).await;
-            self.publish_actor_stopped_events(
+        let claims = actor.claims().context("claims missing")?;
+        self.publish_event(
+            "actor_scaled",
+            event::actor_scaled(
                 claims,
-                &instance.annotations,
+                &actor.annotations,
                 host_id,
-                instance.max_instances,
-                0,
-                &instance.image_reference,
-            )
-            .await?;
-        }
+                0_usize,
+                &actor.image_reference,
+            ),
+        )
+        .await?;
+
         Ok(())
     }
 
     #[instrument(level = "debug", skip_all)]
-    async fn handle_auction_actor(&self, payload: impl AsRef<[u8]>) -> anyhow::Result<Bytes> {
+    async fn handle_auction_actor(
+        &self,
+        payload: impl AsRef<[u8]>,
+    ) -> anyhow::Result<Option<Bytes>> {
         let ActorAuctionRequest {
             actor_ref,
+            actor_id,
             constraints,
         } = serde_json::from_slice(payload.as_ref())
             .context("failed to deserialize actor auction command")?;
 
-        info!(actor_ref, ?constraints, "handling auction for actor");
-
-        let buf = serde_json::to_vec(&ActorAuctionAck {
+        info!(
             actor_ref,
-            constraints,
-            host_id: self.host_key.public_key(),
-        })
-        .context("failed to encode reply")?;
-        Ok(buf.into())
+            actor_id,
+            ?constraints,
+            "handling auction for actor"
+        );
+
+        let host_labels = self.labels.read().await;
+        let constraints_satisfied = constraints
+            .iter()
+            .all(|(k, v)| host_labels.get(k).map(|hv| hv == v).unwrap_or(false));
+        let actor_id_running = self.actors.read().await.contains_key(&actor_id);
+
+        // This host can run the actor if all constraints are satisfied and the actor is not already running
+        if constraints_satisfied && !actor_id_running {
+            let buf = serde_json::to_vec(&ActorAuctionAck {
+                actor_ref,
+                constraints,
```

Comment:
Should we return the actor ID back here as well? Feels like we should because this is a host saying "Yes, I can run actor with $REF and $ID"

Reply:
Yeah that's a good idea, no reason not to